### PR TITLE
Make models work with Marshal.dump

### LINKED ISSUE #494: Cannot Marshal.dump
Will it be possible to Marshal.dump a AR that extend Mobility

## Context

config/initializers/mobility.rb
```ruby
Mobility.configure do 
  plugins do
    backend :table
    active_record
    reader
    writer
    backend_reader
    query
  end  
end
```

app/models/project.rb
```ruby
class Project < ActiveRecord::Base
  extend Mobility
  translates :title
end
```

## Actual Behavior

```ruby
p = Project.new(title: "My title")
Marshal.dump(p)
...
TypeError (can't dump hash with default proc)
```

### PR DESCRIPTION
**Pull request by shioyama**
Fixes #494

Basically, gives names to some stuff that was anonymous.

### CODE REVIEW

**Comment by shioyama** [comment]
This looks like it works. Note that it uses an MD5 hash of the sorted backend options in the backend subclass name, so if a code change changes these options, then Marshal will fail to find the class.

This is a pretty typical problem with Marshal since it dumps everything, so the usual caveats: don't store Marshal-serialized objects in the db and if you're dumping them to `Rails.cache` be careful to clear the cache on deploys that might change cached values (in this case if you change options to `translates`, you'll need to clear any caches that have serialized copies of the objects).

File: lib/mobility/backend.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # frozen-string-literal: true
+require 'digest/md5'
```
Comment (shioyama) [inline lib/mobility/backend.rb:2]: I'm a bit unsure whether this is ok. Any application that uses Mobility will now require md5. I just want a short unique hash for the options hash... :thinking:

Comment (maxime-carbonneau) [reply lib/mobility/backend.rb:2]: I would say it is not a big deal since this is part of Ruby Standard Library.

**Comment by maxime-carbonneau** [comment]
@shioyama Do you have a schedule to release this modification?

**Comment by shioyama** [comment]
@maxime-carbonneau Hey sorry I haven't forgotten about this. It works but I'm not so happy about the md5 thing, I'd like to try to avoid having that in the serialization. I have a vague idea of how to do it differently but just need a bit of time to try.

**Comment by maxime-carbonneau** [comment]
@shioyama I may have miss something, but why don't you use rand or a unique counter instead.

**Comment by shioyama** [comment]
Random stuff or counters won't work because the same configured backend class needs to always return the same name - it has to be deterministic, otherwise when you load it back it won't match and it will fail.

**Comment by maxime-carbonneau** [comment]
Could it be solved by a pseudo md5 : 
<code>
(options.sort.to_s.bytes.each_with_index.sum {|a, b| a + b } % 1_000_000).to_s
</code>
It won't be as "secure" as md5 but it may resolve the issue.

**Comment by shioyama** [comment]
It's not the security that concerns me, it's the arbitrariness of it. I'd like to see if we can avoid assigning a name to the class at all, but I'm not sure that's possible. Give me a couple days to try.

**Comment by shioyama** [comment]
Replaced by #532.

**Pull request closed without merging**